api/tests/test_import_diagnostics.py

```diff
@@ -47,6 +48,46 @@ def test_diagnostics_created(self):
         self.assertEqual(diagnostic.value_sustainable_ht, Decimal("100.1"))
         self.assertIn("seconds", body)
 
+    @authenticate
+    def test_location_found(self):
+        """
+        Test that remaining location information is filled in from import
+        """
+        with open("./api/tests/files/diagnostics_locations.csv") as diag_file:
+            response = self.client.post(reverse("import_diagnostics"), {"file": diag_file})
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        canteen = Canteen.objects.get(siret="21340172201787")
+        self.assertEqual(canteen.city_insee_code, "54318")
+        self.assertEqual(canteen.city, "Liverdun")
+        self.assertEqual(canteen.department, Department.meurthe_et_moselle)
+        # TODO: region?
+        canteen = Canteen.objects.get(siret="73282932000074")
+        self.assertEqual(canteen.postal_code, "07130")
+        self.assertEqual(canteen.city, "Saint-Romain-de-Lerps")
+        self.assertEqual(canteen.department, Department.ardeche)
+        # Given both a city code and postcode, use citycode only to find location
+        canteen = Canteen.objects.get(siret="32441387130915")
+        self.assertEqual(canteen.city_insee_code, "07293")
+        self.assertEqual(canteen.postal_code, "07130")
+        self.assertEqual(canteen.city, "Saint-Romain-de-Lerps")
+        self.assertEqual(canteen.department, Department.ardeche)
+
+    @authenticate
+    def test_location_not_found(self):
+        with open("./api/tests/files/diagnostics_bad_location.csv") as diag_file:
+            response = self.client.post(reverse("import_diagnostics"), {"file": diag_file})
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        canteen = Canteen.objects.get(siret="21340172201787")
+        self.assertEqual(canteen.postal_code, "10")
+        self.assertEqual(canteen.city_insee_code, "")
+        self.assertIsNone(canteen.city)
+        self.assertIsNone(canteen.department)
+
+    # TODO: what if data already exists
+    # TODO: what if API is non responsive
```

Comment:
"# TODO: what if API is non responsive"

I say we give it a reasonable timeout and then act as if not-found so we leave everything empty. Logging would be good as well to realise how many timeouts we get.

Reply:
Oh yeah, I could've phrased the comment better, I meant more "TODO: write a test for unresponsive API" or something so that the timeout that is set is tested